sailfishusermanager.cpp

```diff
@@ -348,3 +371,97 @@ bool SailfishUserManager::hasAccessRights(uint uid_to_modify) {
 
     return true;
 }
+
+void SailfishUserManager::switchUser(uint uid)
+{
+    if (checkCallerUid() == SAILFISH_UNDEFINED_UID)
+        return;
+
+    if (m_switchUser) {
+        auto message = QStringLiteral("Already switching user");
+        qCWarning(lcSUM) << message;
+        sendErrorReply(QDBusError::Failed, message);
+        return;
```

Comment:
What's D-Bus client supposed to do in this case? Retry? What if `m_switchUser` equals `uid` - that's not really an error, such (essentially, duplicate) request can be completed together with the one currently being processed.

I personally prefer to queue incoming requests and complete them one after another if they can't be done in parallel. That's simpler for the client (no need to implement retry logic) and better from the performance point of view (avoiding flood of retries). But of course that's slightly more complicated than handling one request at a time, so it may be too much to ask.

Reply:
And if smart handling of incoming requests is too much to ask, then at least service should return a special error in this case (`Busy` or something), so that a smart client could insert a delay before retrying the request.

Reply:
Introducing service specific errors, should be enough as there's only one client (settings UI) for this service.